evm/src/main/java/org/hyperledger/besu/evm/fluent/SimpleWorld.java

```diff
@@ -55,13 +55,15 @@ public WorldUpdater updater() {
 
   @Override
   public Account get(final Address address) {
+    Optional<SimpleAccount> account = Optional.empty();
     if (accounts.containsKey(address)) {
-      return accounts.get(address);
+      account = accounts.get(address);
     } else if (parent != null) {
-      return parent.get(address);
-    } else {
-      return null;
+      if (parent.get(address) instanceof SimpleAccount accountFromParent) {
+        account = Optional.of(accountFromParent);
+      }
```

Comment:
Unless I am missing something, since parent is SimpleWorld, it can only return a SimpleAccount. is this a different flavor of not null check?

Reply:
It's needed there for recursion and  since `parent.get()` returns an `Account`, even though it's always backed by a `SimpleAccount`. The `instanceof` does the null check and the cast in one go.